expectation.go

```diff
@@ -41,6 +41,12 @@ type Expectation struct {
 	// Times expected to be called
 	minCalls, maxCalls int
 
+	// stdin expectation, as a string or a Matcher
+	stdin interface{}
```

Comment:
I don’t love the Interface{} type here, is there precedent elsewhere in the API?

Reply:
Yeah I'm not sure how else to handle `string` or `Matcher`.
It's the same as arguments;
https://github.com/buildkite/bintest/blob/eac69ec698ff63e588c85fbe390e4a18ca46311a/mock.go#L195

They're only really different because they travel in packs, in a `type Arguments` which is really just `[]interface{}`.

Reply:
Yeah, meh.